trendmicro_deepsecurity/help.md

```diff
@@ -0,0 +1,141 @@
+
+# Key Features
+
+* Utilize the Deep Security IPS engine to protect against vulnerabilities without installing a patch (virtual patching)
+
+# Requirements
+
+* Trend Micro Deep Security
+
+# Documentation
+
+## Setup
+
+* [Create an API key for Deep Security](https://help.deepsecurity.trendmicro.com/api-key.html)
+* Create a new connection and enter the Deep Security Manager URL and API key 
+
+The connection configuration accepts the following parameters:
+
+|Name|Type|Default|Required|Description|Enum|
+|----|----|-------|--------|-----------|----|
+|dsm_api_key|credential_secret_key|None|True|API key of the Deep Security Manager|None|
+|dsm_url|string|https://app.deepsecurity.trendmicro.com|True|URL of the Deep Security Manager|None|
+
+## Technical Details
+
+### Actions
+
+#### Search
+
+Search for matching IPS rules in Deep Security by CVE ID
+
+##### Input
+
+|Name|Type|Default|Required|Description|Enum|
+|----|----|-------|--------|-----------|----|
+|vulnerabilities|[]string|None|True|CVEs to protect against|None|
+
+##### Output
+
+|Name|Type|Required|Description|
+|----|----|--------|-----------|
+|ips_rules|[]integer|False|IPS rules matching the given CVEs|
+|matched_cves|[]string|False|CVEs with at least one matching IPS rule|
+|missed_cves|[]string|False|CVEs without matching IPS rules|
+
+Example output:
+
+```
+{
+  "ips_rules": [
+    108,
+    6348,
+    3317,
+    3318,
+    2874,
+    2875,
+    2876
+  ],
+  "matched_cves": [
+    "CVE-2005-0045",
+    "CVE-2017-0144",
+    "CVE-2014-0160"
+  ],
+  "missed_cves": [
+    "CVE-1337-1337"
+  ]
+}
+```
+
+#### Deploy
+
+Deploy the given Deep Security IPS rules to a computer or policy
+
+##### Input
+
+|Name|Type|Default|Required|Description|Enum|
+|----|----|-------|--------|-----------|----|
+|computer_or_policy|string|None|True|Target for rule assignment|['computer', 'policy']|
+|id|integer|None|True|ID of the target asset or policy|None|
+|rules|[]integer|None|True|IPS rules to assign|None|
+
+##### Output
+
+|Name|Type|Required|Description|
+|----|----|--------|-----------|
+|rules_assigned|[]integer|False|All rules currently assigned to the asset|
+|rules_not_assigned|[]integer|False|Unassigned rules|
+
+Example output:
+
+```
+{
+  "rules_assigned": [
+    108,
+    2874,
+    2875,
+    2876,
+    3118,
+    3124,
+    3125,
+    3126,
+    3317,
+    3318,
+    4593,
+    5177,
+    5892,
+    6348,
+    6745
+  ],
+  "rules_not_assigned": []
+}
+```
+
+### Triggers
+
+_This plugin does not contain any triggers._
+
+### Custom Output Types
+
+_This plugin does not contain any custom output types._
+
+## Troubleshooting
+
+_This plugin does not contain any troubleshooting information._
+
+# Version History
+
+* 1.0.0 - Initial plugin
+
+# Links
+
+## References
+
+* [Trend Micro Deep Security](https://www.trendmicro.com/en_us/business/products/hybrid-cloud/deep-security.html)
+
+## Author
```

Comment:
This is not a valid section for extensions. See another extension's help.md for the format.